<?xml version='1.0' encoding='UTF-8'?>
<robot name="orchard_ugv">
    <!-- Constants -->
    <!-- M_PI is not directly used in URDF, so we can replace it with its value where needed -->

    <!-- Insert Robot Content -->
    <link name="base_footprint"/>
    <link name="base_link">
        <visual>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <geometry>
                <mesh filename="/root/ros2_ws/src/Isaac_ml/meshes/kimm_orchard/visual/base_link.dae" scale="1 1 1"/>
            </geometry>
        </visual>
        <collision>
            <origin xyz="0 0 0.2" rpy="0 0 0"/>
            <geometry>
                <box size="1.6 0.8 0.8"/>
            </geometry>
        </collision>
        <inertial>
            <origin xyz="-0.034 0.000 0.256" rpy="0 0 0"/>
            <mass value="273.0"/>
            <inertia ixx="35.7508" ixy="0.0012" ixz="2.5504" iyy="70.3506" iyz="-0.0008" izz="70.8845"/>
        </inertial>
    </link>

    <joint name="base_link_fixed_joint" type="fixed">
        <parent link="base_footprint"/>
        <child link="base_link"/>
        <origin xyz="0 0 0.5" rpy="0 0 0"/>
    </joint>

    <link name="l_bogie_link">
        <visual>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <geometry>
                <mesh filename="/root/ros2_ws/src/Isaac_ml/meshes/kimm_orchard/visual/l_bogie_link.dae" scale="1 1 1"/>
            </geometry>
        </visual>
        <collision>
            <origin xyz="0 0.045 0.03" rpy="0 0 0"/>
            <geometry>
                <box size="1.5 0.045 0.095"/>
            </geometry>
        </collision>
        <inertial>
            <origin xyz="-0.081 0.104 0.021" rpy="0 0 0"/>
            <mass value="15.0"/>
            <inertia ixx="0.0772" ixy="-0.0134" ixz="0.0193" iyy="2.8925" iyz="0.0002" izz="2.9208"/>
        </inertial>
    </link>

    <!-- Assuming is_sim is false for this example -->
    <joint name="l_bogie_joint" type="revolute">
        <parent link="base_link"/>
        <child link="l_bogie_link"/>
        <origin xyz="0 0.4174 0" rpy="0 0 0"/>
        <axis xyz="0 1 0"/>
        <limit effort="100" velocity="1.0" lower="-0.5236" upper="0.5236"/> <!-- -30*pi/180 to 30*pi/180 -->
    </joint>

    <link name="l_front_wheel_rotate_link">
        <visual>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <geometry>
                <mesh filename="/root/ros2_ws/src/Isaac_ml/meshes/kimm_orchard/visual/l_wheel_rotate_link.dae" scale="1 1 1"/>
            </geometry>
        </visual>
        <collision>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <geometry>
                <cylinder radius="0.08" length="0.05"/>
            </geometry>
        </collision>
        <inertial>
            <origin xyz="-0.045 -0.095 -0.153" rpy="0 0 0"/>
            <mass value="10.5"/>
            <inertia ixx="0.1274" ixy="-0.0189" ixz="0.0011" iyy="0.1229" iyz="-0.0173" izz="0.1128"/>
        </inertial>
    </link>

    <joint name="l_front_wheel_rotate_joint" type="revolute">
        <parent link="l_bogie_link"/>
        <child link="l_front_wheel_rotate_link"/>
        <origin xyz="0.625 0.141 -0.0431345" rpy="0 0 0"/>
        <axis xyz="0 0 1"/>
        <limit effort="100" velocity="2.0" lower="-1.7453" upper="1.7453"/> <!-- -100*pi/180 to 100*pi/180 -->
    </joint>

    <link name="l_front_wheel">
        <visual>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <geometry>
                <mesh filename="/root/ros2_ws/src/Isaac_ml/meshes/kimm_orchard/visual/l_wheel_link.dae" scale="1 1 1"/>
            </geometry>
        </visual>
        <collision>
            <origin xyz="0 0 0" rpy="1.5708 0 0"/> <!-- pi/2 -->
            <geometry>
                <cylinder radius="0.2" length="0.105"/>
            </geometry>
        </collision>
        <inertial>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <mass value="3.0"/>
            <inertia ixx="0.0306" ixy="0.0" ixz="0.0" iyy="0.0559" iyz="0.0" izz="0.0306"/>
        </inertial>
    </link>

    <joint name="l_front_wheel_joint" type="continuous">
        <parent link="l_front_wheel_rotate_link"/>
        <child link="l_front_wheel"/>
        <origin xyz="0 0 -0.257" rpy="0 0 0"/>
        <axis xyz="0 1 0"/>
    </joint>

    <link name="l_rear_wheel_rotate_link">
        <visual>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <geometry>
                <mesh filename="/root/ros2_ws/src/Isaac_ml/meshes/kimm_orchard/visual/l_wheel_rotate_link.dae" scale="1 1 1"/>
            </geometry>
        </visual>
        <collision>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <geometry>
                <cylinder radius="0.08" length="0.05"/>
            </geometry>
        </collision>
        <inertial>
            <origin xyz="-0.045 -0.095 -0.153" rpy="0 0 0"/>
            <mass value="10.5"/>
            <inertia ixx="0.1274" ixy="-0.0189" ixz="0.0011" iyy="0.1229" iyz="-0.0173" izz="0.1128"/>
        </inertial>
    </link>

    <joint name="l_rear_wheel_rotate_joint" type="revolute">
        <parent link="l_bogie_link"/>
        <child link="l_rear_wheel_rotate_link"/>
        <origin xyz="-0.625 0.141 -0.0431345" rpy="0 0 0"/>
        <axis xyz="0 0 1"/>
        <limit effort="100" velocity="2.0" lower="-1.7453" upper="1.7453"/> <!-- -100*pi/180 to 100*pi/180 -->
    </joint>

    <link name="l_rear_wheel">
        <visual>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <geometry>
                <mesh filename="/root/ros2_ws/src/Isaac_ml/meshes/kimm_orchard/visual/l_wheel_link.dae" scale="1 1 1"/>
            </geometry>
        </visual>
        <collision>
            <origin xyz="0 0 0" rpy="1.5708 0 0"/> <!-- pi/2 -->
            <geometry>
                <cylinder radius="0.2" length="0.105"/>
            </geometry>
        </collision>
        <inertial>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <mass value="3.0"/>
            <inertia ixx="0.0306" ixy="0.0" ixz="0.0" iyy="0.0559" iyz="0.0" izz="0.0306"/>
        </inertial>
    </link>

    <joint name="l_rear_wheel_joint" type="continuous">
        <parent link="l_rear_wheel_rotate_link"/>
        <child link="l_rear_wheel"/>
        <origin xyz="0 0 -0.257" rpy="0 0 0"/>
        <axis xyz="0 1 0"/>
    </joint>

    <link name="r_bogie_link">
        <visual>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <geometry>
                <mesh filename="/root/ros2_ws/src/Isaac_ml/meshes/kimm_orchard/visual/r_bogie_link.dae" scale="1 1 1"/>
            </geometry>
        </visual>
        <collision>
            <origin xyz="0 -0.045 0.03" rpy="0 0 0"/>
            <geometry>
                <box size="1.5 0.045 0.095"/>
            </geometry>
        </collision>
        <inertial>
            <origin xyz="-0.071 -0.103 0.021" rpy="0 0 0"/>
            <mass value="15.0"/>
            <inertia ixx="0.0766" ixy="0.0245" ixz="0.0182" iyy="2.8138" iyz="-0.0003" izz="2.8416"/>
        </inertial>
    </link>

    <joint name="r_bogie_joint" type="revolute">
        <parent link="base_link"/>
        <child link="r_bogie_link"/>
        <origin xyz="0 -0.4174 0" rpy="0 0 0"/>
        <axis xyz="0 1 0"/>
        <limit effort="100" velocity="1.0" lower="-0.5236" upper="0.5236"/> <!-- -30*pi/180 to 30*pi/180 -->
    </joint>

    <link name="r_front_wheel_rotate_link">
        <visual>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <geometry>
                <mesh filename="/root/ros2_ws/src/Isaac_ml/meshes/kimm_orchard/visual/r_wheel_rotate_link.dae" scale="1 1 1"/>
            </geometry>
        </visual>
        <collision>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <geometry>
                <cylinder radius="0.08" length="0.05"/>
            </geometry>
        </collision>
        <inertial>
            <origin xyz="-0.045 0.096 -0.155" rpy="0 0 0"/>
            <mass value="10.5"/>
            <inertia ixx="0.1282" ixy="0.0191" ixz="0.0012" iyy="0.1229" iyz="0.0174" izz="0.1136"/>
        </inertial>
    </link>

    <joint name="r_front_wheel_rotate_joint" type="revolute">
        <parent link="r_bogie_link"/>
        <child link="r_front_wheel_rotate_link"/>
        <origin xyz="0.625 -0.141 -0.0431345" rpy="0 0 0"/>
        <axis xyz="0 0 1"/>
        <limit effort="100" velocity="2.0" lower="-1.7453" upper="1.7453"/> <!-- -100*pi/180 to 100*pi/180 -->
    </joint>

    <link name="r_front_wheel">
        <visual>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <geometry>
                <mesh filename="/root/ros2_ws/src/Isaac_ml/meshes/kimm_orchard/visual/r_wheel_link.dae" scale="1 1 1"/>
            </geometry>
        </visual>
        <collision>
            <origin xyz="0 0 0" rpy="1.5708 0 0"/> <!-- pi/2 -->
            <geometry>
                <cylinder radius="0.2" length="0.105"/>
            </geometry>
        </collision>
        <inertial>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <mass value="3.0"/>
            <inertia ixx="0.0306" ixy="0.0" ixz="0.0" iyy="0.0560" iyz="0.0" izz="0.0306"/>
        </inertial>
    </link>

    <joint name="r_front_wheel_joint" type="continuous">
        <parent link="r_front_wheel_rotate_link"/>
        <child link="r_front_wheel"/>
        <origin xyz="0 0 -0.257" rpy="0 0 0"/>
        <axis xyz="0 1 0"/>
    </joint>

    <link name="r_rear_wheel_rotate_link">
        <visual>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <geometry>
                <mesh filename="/root/ros2_ws/src/Isaac_ml/meshes/kimm_orchard/visual/r_wheel_rotate_link.dae" scale="1 1 1"/>
            </geometry>
        </visual>
        <collision>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <geometry>
                <cylinder radius="0.08" length="0.05"/>
            </geometry>
        </collision>
        <inertial>
            <origin xyz="-0.045 0.096 -0.155" rpy="0 0 0"/>
            <mass value="10.5"/>
            <inertia ixx="0.1282" ixy="0.0191" ixz="0.0012" iyy="0.1229" iyz="0.0174" izz="0.1136"/>
        </inertial>
    </link>

    <joint name="r_rear_wheel_rotate_joint" type="revolute">
        <parent link="r_bogie_link"/>
        <child link="r_rear_wheel_rotate_link"/>
        <origin xyz="-0.625 -0.141 -0.0431345" rpy="0 0 0"/>
        <axis xyz="0 0 1"/>
        <limit effort="100" velocity="2.0" lower="-1.7453" upper="1.7453"/> <!-- -100*pi/180 to 100*pi/180 -->
    </joint>

    <link name="r_rear_wheel">
        <visual>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <geometry>
                <mesh filename="/root/ros2_ws/src/Isaac_ml/meshes/kimm_orchard/visual/r_wheel_link.dae" scale="1 1 1"/>
            </geometry>
        </visual>
        <collision>
            <origin xyz="0 0 0" rpy="1.5708 0 0"/> <!-- pi/2 -->
            <geometry>
                <cylinder radius="0.2" length="0.105"/>
            </geometry>
        </collision>
        <inertial>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <mass value="3.0"/>
            <inertia ixx="0.0306" ixy="0.0" ixz="0.0" iyy="0.0560" iyz="0.0" izz="0.0306"/>
        </inertial>
    </link>

    <joint name="r_rear_wheel_joint" type="continuous">
        <parent link="r_rear_wheel_rotate_link"/>
        <child link="r_rear_wheel"/>
        <origin xyz="0 0 -0.257" rpy="0 0 0"/>
        <axis xyz="0 1 0"/>
    </joint>

    <link name="roll_balance_link">
        <visual>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <geometry>
                <mesh filename="/root/ros2_ws/src/Isaac_ml/meshes/kimm_orchard/visual/roll_balance_link.dae" scale="1 1 1"/>
            </geometry>
        </visual>
        <collision>
            <origin xyz="-0.007 0 0" rpy="0 0 0"/>
            <geometry>
                <box size="0.024 0.940 0.098"/>
            </geometry>
        </collision>
        <inertial>
            <origin xyz="-0.007 0.000 0.000" rpy="0 0 0"/>
            <mass value="1.0"/>
            <inertia ixx="0.0598" ixy="0.0" ixz="0.0" iyy="0.0006" iyz="0.0" izz="0.0593"/>
        </inertial>
    </link>

    <joint name="roll_balance_joint" type="revolute">
        <parent link="base_link"/>
        <child link="roll_balance_link"/>
        <origin xyz="-0.853 0 0.637" rpy="0 0 0"/>
        <axis xyz="1 0 0"/>
        <limit effort="100" velocity="1.0" lower="-0.5236" upper="0.5236"/> <!-- -30*pi/180 to 30*pi/180 -->
    </joint>

    <link name="l_roll_static_link">
        <visual>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <geometry>
                <mesh filename="/root/ros2_ws/src/Isaac_ml/meshes/kimm_orchard/visual/l_roll_static_link.dae" scale="1 1 1"/>
            </geometry>
        </visual>
        <collision>
            <origin xyz="0.03 0 -0.3" rpy="0 0 0"/>
            <geometry>
                <cylinder radius="0.025" length="0.4"/>
            </geometry>
        </collision>
        <inertial>
            <origin xyz="0.012 0.003 -0.225" rpy="0 0 0"/>
            <mass value="1.0"/>
            <inertia ixx="0.0258" ixy="0.0" ixz="0.0006" iyy="0.0261" iyz="0.0004" izz="0.0005"/>
        </inertial>
    </link>

    <joint name="l_roll_static_joint" type="revolute">
        <parent link="roll_balance_link"/>
        <child link="l_roll_static_link"/>
        <origin xyz="0.005 0.445 0" rpy="0 0 0"/>
        <axis xyz="1 0 0"/>
        <limit effort="100" velocity="1.0" lower="-0.5236" upper="0.5236"/> <!-- -30*pi/180 to 30*pi/180 -->
        <mimic joint="roll_balance_joint" multiplier="-1.0" offset="0.0"/>
    </joint>

    <link name="r_roll_static_link">
        <visual>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <geometry>
                <mesh filename="/root/ros2_ws/src/Isaac_ml/meshes/kimm_orchard/visual/r_roll_static_link.dae" scale="1 1 1"/>
            </geometry>
        </visual>
        <collision>
            <origin xyz="0.03 0 -0.3" rpy="0 0 0"/>
            <geometry>
                <cylinder radius="0.025" length="0.4"/>
            </geometry>
        </collision>
        <inertial>
            <origin xyz="0.012 -0.003 -0.225" rpy="0 0 0"/>
            <mass value="1.0"/>
            <inertia ixx="0.0258" ixy="0.0" ixz="0.0006" iyy="0.0261" iyz="-0.0004" izz="0.0005"/>
        </inertial>
    </link>

    <joint name="r_roll_static_joint" type="revolute">
        <parent link="roll_balance_link"/>
        <child link="r_roll_static_link"/>
        <origin xyz="0.005 -0.445 0" rpy="0 0 0"/>
        <axis xyz="1 0 0"/>
        <limit effort="100" velocity="1.0" lower="-0.5236" upper="0.5236"/> <!-- -30*pi/180 to 30*pi/180 -->
        <mimic joint="roll_balance_joint" multiplier="-1.0" offset="0.0"/>
    </joint>

    <link name="arm_mount_link"/>
    <joint name="arm_mount_fixed_joint" type="fixed">
        <parent link="base_link"/>
        <child link="arm_mount_link"/>
        <origin xyz="0.404 0 0.555" rpy="0 0 0"/>
    </joint>

    <link name="lidar_mount_link">
        <visual>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <geometry>
                <mesh filename="/root/ros2_ws/src/Isaac_ml/meshes/kimm_orchard/visual/ouster_os0.dae" scale="1 1 1"/>
            </geometry>
        </visual>
        <collision>
            <origin xyz="0 0 0.0375" rpy="0 0 0"/>
            <geometry>
                <cylinder radius="0.0425" length="0.075"/>
            </geometry>
        </collision>
        <inertial>
            <origin xyz="0 0 0.0375" rpy="0 0 0"/>
            <mass value="0.5"/>
            <inertia ixx="0.0009" ixy="0" ixz="0" iyy="0.0009" iyz="0" izz="0.0009"/> <!-- Example inertia values -->
        </inertial>
    </link>
        <link name="arm1_link">
            <inertial>
                <origin xyz="3.59244523089111e-15 0.404 -0.517999999999538" rpy="0.0 -0.0 0.0"/>
                <mass value="0.0"/>
                <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
            </inertial>
            <visual>
                <geometry>
                    <mesh filename="/root/ros2_ws/src/Isaac_ml/meshes/kimm_orchard/visual/arm1_link.dae"/>
                </geometry>
                <geometry>
                    <mesh filename="/root/ros2_ws/src/Isaac_ml/meshes/kimm_orchard/visual/r_roll_static_link.dae" scale="1 1 1"/>
                </geometry>
                <material name=""/>
                <origin xyz="3.59244523089111e-15 0.404 -0.517999999999538" rpy="1.5707963267949003 -6.70878395900397e-16 3.87006573320564e-31"/>
            </visual>
            <collision>
                <geometry>
                    <mesh filename="/root/ros2_ws/src/Isaac_ml/meshes/kimm_orchard/visual/arm1_link.dae"/>
                </geometry>
                <origin xyz="3.59244523089111e-15 0.404 -0.517999999999538" rpy="1.5707963267949003 -6.70878395900397e-16 3.87006573320564e-31"/>
            </collision>
        </link>
        <link name="arm2_link">
            <inertial>
                <origin xyz="-0.343235015894633 -0.589043999999388 0.120999420094225" rpy="0.0 -0.0 0.0"/>
                <mass value="10.905"/>
                <inertia ixx="0.123531144775942" ixy="-0.00167187632637416" ixz="0.0147981043711083" iyy="0.12556" iyz="-0.0109653944497318" izz="0.0287598552240577"/>
            </inertial>
            <visual>
                <geometry>
                    <mesh filename="/root/ros2_ws/src/Isaac_ml/meshes/kimm_orchard/visual/arm2_link.dae"/>
                </geometry>
                <material name=""/>
                <origin xyz="-0.400497772647027 -0.616999999999388 -0.141580449364795" rpy="3.93709538557874e-15 1.43902809440512 3.90296517976033e-15"/>
            </visual>
            <collision>
                <geometry>
                    <mesh filename="/root/ros2_ws/src/Isaac_ml/meshes/kimm_orchard/visual/arm2_link.dae"/>
                </geometry>
                <origin xyz="-0.400497772647027 -0.616999999999388 -0.141580449364795" rpy="3.93709538557874e-15 1.43902809440512 3.90296517976033e-15"/>
            </collision>
        </link>
        <link name="arm3_link">
            <inertial>
                <origin xyz="-0.23746464444949 -0.581800538201215 -0.0932612938687763" rpy="0.0 -0.0 0.0"/>
                <mass value="5.1715"/>
                <inertia ixx="0.0620232894908731" ixy="0.0027594891181694" ixz="0.00658449065099018" iyy="0.054662" iyz="0.019771837993639" izz="0.0157657105091269"/>
            </inertial>
            <visual>
                <geometry>
                    <mesh filename="/root/ros2_ws/src/Isaac_ml/meshes/kimm_orchard/visual/arm3_link.dae"/>
                </geometry>
                <material name=""/>
                <origin xyz="-0.208487038807708 -0.684370538201215 0.159880449364673" rpy="-3.14159265358978 -1.43902809440512 3.14159265358978"/>
            </visual>
            <collision>
                <geometry>
                    <mesh filename="/root/ros2_ws/src/Isaac_ml/meshes/kimm_orchard/visual/arm3_link.dae"/>
                </geometry>
                <origin xyz="-0.208487038807708 -0.684370538201215 0.159880449364673" rpy="-3.14159265358978 -1.43902809440512 3.14159265358978"/>
            </collision>
        </link>
        <link name="arm4_link">
            <inertial>
                <origin xyz="-0.272687359398989 -0.893175042124194 -0.153736266004766" rpy="0.0 -0.0 0.0"/>
                <mass value="0.86844"/>
                <inertia ixx="0.00100159971746297" ixy="-0.000254319066847665" ixz="-4.85414251946025e-06" iyy="0.000885589999999999" iyz="3.64996945019128e-05" izz="0.000982180282537027"/>
            </inertial>
            <visual>
                <geometry>
                    <mesh filename="/root/ros2_ws/src/Isaac_ml/meshes/kimm_orchard/visual/arm4_link.dae"/>
                </geometry>
                <material name=""/>
                <origin xyz="-0.318109405184127 -0.902927642124194 -0.159480449364736" rpy="1.1517030239331e-14 1.43902809440512 1.14171904909914e-14"/>
            </visual>
            <collision>
                <geometry>
                    <mesh filename="/root/ros2_ws/src/Isaac_ml/meshes/kimm_orchard/visual/arm4_link.dae"/>
                </geometry>
                <origin xyz="-0.318109405184127 -0.902927642124194 -0.159480449364736" rpy="1.1517030239331e-14 1.43902809440512 1.14171904909914e-14"/>
            </collision>
        </link>
        <link name="arm5_link">
            <inertial>
                <origin xyz="0.202541583766248 0.24718018200937 -0.987196222269241" rpy="0.0 -0.0 0.0"/>
                <mass value="0.86844"/>
                <inertia ixx="0.000983859499110623" ixy="-1.83654467228935e-07" ixz="8.95386689563206e-07" iyy="0.00101420158979828" iyz="0.000253427259349587" izz="0.000871318911091097"/>
            </inertial>
            <visual>
                <geometry>
                    <mesh filename="/root/ros2_ws/src/Isaac_ml/meshes/kimm_orchard/visual/arm5_link.dae"/>
                </geometry>
                <material name=""/>
                <origin xyz="0.209830449364735 0.292651048122776 -0.978797276250015" rpy="1.54297911628478 -0.0036862949803813196 -0.131716957803607"/>
            </visual>
            <collision>
                <geometry>
                    <mesh filename="/root/ros2_ws/src/Isaac_ml/meshes/kimm_orchard/visual/arm5_link.dae"/>
                </geometry>
                <origin xyz="0.209830449364735 0.292651048122776 -0.978797276250015" rpy="1.54297911628478 -0.0036862949803813196 -0.131716957803607"/>
            </collision>
        </link>
        <link name="arm6_link">
            <inertial>
                <origin xyz="0.165663052283443 -1.02932559947292 -0.363435560653319" rpy="0.0 -0.0 0.0"/>
                <mass value="0.28425"/>
                <inertia ixx="0.000184030011635817" ixy="-1.94361694308924e-09" ixz="3.17188458993215e-08" iyy="0.000187895522250448" iyz="2.32746985867529e-06" izz="0.000270494466113735"/>
            </inertial>
            <visual>
                <geometry>
                    <mesh filename="/root/ros2_ws/src/Isaac_ml/meshes/kimm_orchard/visual/arm6_link.dae"/>
                </geometry>
                <material name=""/>
                <origin xyz="0.165655145892847 -1.02976665535819 -0.385036648519622" rpy="-0.0278158578089435 0.00036695245300989195 -1.02097303261102e-05"/>
            </visual>
            <collision>
                <geometry>
                    <mesh filename="/root/ros2_ws/src/Isaac_ml/meshes/kimm_orchard/visual/arm6_link.dae"/>
                </geometry>
                <origin xyz="0.165655145892847 -1.02976665535819 -0.385036648519622" rpy="-0.0278158578089435 0.00036695245300989195 -1.02097303261102e-05"/>
            </collision>
        </link>
    <joint name="lidar_mount_fixed_joint" type="fixed">
        <parent link="base_link"/>
        <child link="lidar_mount_link"/>
        <origin xyz="0.7012 0 0.6155" rpy="0 0 0"/>
    </joint>

    <link name="lidar_link"/>
    <joint name="lidar_fixed_joint" type="fixed">
        <parent link="lidar_mount_link"/>
        <child link="lidar_link"/>
        <origin xyz="0 0 0.038195" rpy="0 0 0"/>
    </joint>

    <link name="rgbd_camera_mount_link"/>
    <joint name="rgbd_camera_mount_fixed_joint" type="fixed">
        <parent link="base_link"/>
        <child link="rgbd_camera_mount_link"/>
        <origin xyz="0.763 0 0.578" rpy="0 0 0"/>
    </joint>

    <link name="front_camera_mount_link"/>
    <joint name="front_camera_mount_fixed_joint" type="fixed">
        <parent link="base_link"/>
        <child link="front_camera_mount_link"/>
        <origin xyz="0.800 0 0.4734" rpy="0 0 0"/>
    </joint>

    <link name="rear_camera_mount_link"/>
    <joint name="rear_camera_mount_fixed_joint" type="fixed">
        <parent link="base_link"/>
        <child link="rear_camera_mount_link"/>
        <origin xyz="-0.800 0 0.4734" rpy="0 0 3.14159"/> <!-- pi -->
    </joint>

    <link name="left_camera_mount_link"/>
    <joint name="left_camera_mount_fixed_joint" type="fixed">
        <parent link="base_link"/>
        <child link="left_camera_mount_link"/>
        <origin xyz="0 0.403 0.4734" rpy="0 0 1.5708"/> <!-- pi/2 -->
    </joint>

    <link name="right_camera_mount_link"/>
    <joint name="right_camera_mount_fixed_joint" type="fixed">
        <parent link="base_link"/>
        <child link="right_camera_mount_link"/>
        <origin xyz="0 -0.403 0.4734" rpy="0 0 -1.5708"/> <!-- -pi/2 -->
    </joint>

    <joint name="arm1_joint" type="continuous">
        <limit effort="0.0" velocity="0.0"/>
        <parent link="arm_mount_link"/>
        <child link="arm1_link"/>
        <axis xyz="0. 0. 1."/>
        <origin xyz="0.0 0.0 0.08" rpy="0.0 -0.0 1.5708"/>
    </joint>
    <joint name="arm2_joint" type="continuous">
        <limit effort="0.0" velocity="0.0"/>
        <parent link="arm1_link"/>
        <child link="arm2_link"/>
        <axis xyz="0. 0. 1."/>
        <origin xyz="-0.087733 -0.011628 0.099" rpy="1.5708 -0.0 -1.5708"/>
    </joint>
    <joint name="arm3_joint" type="continuous">
        <limit effort="0.0" velocity="0.0"/>
        <parent link="arm2_link"/>
        <child link="arm3_link"/>
        <axis xyz="0. 0. 1."/>
        <origin xyz="-0.60898 0.067371 0.0183" rpy="3.1415853071795867 -0.0 -3.1415853071795867"/>
    </joint>
    <joint name="arm4_joint" type="continuous">
        <limit effort="0.0" velocity="0.0"/>
        <parent link="arm3_link"/>
        <child link="arm4_link"/>
        <axis xyz="0. 0. 1."/>
        <origin xyz="-0.5266 0.21856 0.0004" rpy="-3.1415853071795867 -0.0 -3.1415853071795867"/>
    </joint>
    <joint name="arm5_joint" type="continuous">
        <limit effort="0.0" velocity="0.0"/>
        <parent link="arm4_link"/>
        <child link="arm5_link"/>
        <axis xyz="0. 0. 1."/>
        <origin xyz="0.0018882 0.067274 0.05035" rpy="-1.9127926535897932 1.5707926535897934 -0.36899265358979305"/>
    </joint>
    <joint name="arm6_joint" type="continuous">
        <limit effort="0.0" velocity="0.0"/>
        <parent link="arm5_link"/>
        <child link="arm6_link"/>
        <axis xyz="0. 0. 1."/>
        <origin xyz="-0.00886342266060644 -0.0667137897194216 0.0503499999999968" rpy="1.5707963267949003 -0.00369650495889594 -0.13208391274984796"/>
    </joint>
</robot>